File: SretneSapice/SretneSapice.Services/Database/DogWalkerLocation.cs
using System;
using System.Collections.Generic;

namespace SretneSapice.Services.Database;

public partial class DogWalkerLocation
{
    public int DogWalkerId { get; set; }

    public DateTime Timestamp { get; set; }

    public double? Latitude { get; set; }

    public double? Longitude { get; set; }

    public virtual DogWalker DogWalker { get; set; } = null!;
}

File: SretneSapice/UI/sretnesapice_mobile/lib/models/dog_walker_location.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sretnesapice_mobile/models/dog_walker.dart';

part 'dog_walker_location.g.dart';

@JsonSerializable()
class DogWalkerLocation {
  int? dogWalkerId;
  DateTime? timestamp;
  double? latitude;
  double? longitude;
  DogWalker? dogWalker;

  DogWalkerLocation(
    this.dogWalkerId,
    this.timestamp,
    this.latitude,
    this.longitude,
    this.dogWalker,
  );

  factory DogWalkerLocation.fromJson(Map<String, dynamic> json) =>
      _$DogWalkerLocationFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$DogWalkerLocationToJson(this);
}
